Nadim.CinemaReservationSystem.Web/CustomExceptionMiddleware.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+
+namespace Nadim.CinemaReservationSystem.Web
+{
+    // You may need to install the Microsoft.AspNetCore.Http.Abstractions package into your project
+    public class CustomExceptionMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public CustomExceptionMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            try
+            {
+                await _next(httpContext);
+            }
+            catch
+            {
+                await HandleExceptionAsync(httpContext);
+            }
+        }
+        private static Task HandleExceptionAsync(HttpContext context)
+        {
+            context.Response.ContentType = "application/json";
+            context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+
+            return context.Response.WriteAsync(JsonConvert.SerializeObject(
+                new Result {
+                    ResultOk = false,
+                    Details="Server side error."
+                }
+            ));
+        }
+    }
+
+    // Extension method used to add the middleware to the HTTP request pipeline.
+    public static class CustomExceptionMiddlewareExtensions
+    {
+        public static IApplicationBuilder UseCustomExceptionMiddleware(this IApplicationBuilder builder)
```

Comment:
You don't need Middleware part in method name. Per ASP.NET Core naming conventions Use in method name already suggests that this method will add middleware to the pipeline.